identity/src/keypair.rs

```diff
@@ -360,19 +360,21 @@ impl Keypair {
     /// ```
     #[allow(unused_variables, unreachable_code)]
     pub fn derive_secret(&self, domain: &[u8]) -> Option<[u8; 32]> {
-        #[cfg(any(
+        #[cfg(not(any(
             feature = "ecdsa",
             feature = "secp256k1",
             feature = "ed25519",
             feature = "rsa"
-        ))]
-        return Some(
-            hkdf::Hkdf::<sha2::Sha256>::extract(None, &[domain, &self.secret()?].concat())
-                .0
-                .into(),
-        );
+        )))]
+        return None;
+
+        let hkdf =
+            hkdf::Hkdf::<sha2::Sha256>::from_prk(&self.secret()?).expect("secret to have 32 bytes");
```

Comment:
Why `from_prk` and not the default `new`?

From the documentation:

> If you already have a strong key to work from (uniformly-distributed and long enough), you can save a tiny amount of time by skipping the extract step. In this case, you pass a Pseudo-Random Key (PRK) into the [Hkdf::from_prk](https://docs.rs/hkdf/latest/hkdf/struct.Hkdf.html#method.from_prk) constructor, then use the resulting [Hkdf](https://docs.rs/hkdf/latest/hkdf/struct.Hkdf.html) object as usual.

Performance is irrelevant here. Why not follow the default way of doing things?

Again, coming from a novice.

Reply:
> > If you already have a strong key to work from (uniformly-distributed and long enough), you can save a tiny amount of time by skipping the extract step. In this case, you pass a Pseudo-Random Key (PRK) into the [Hkdf::from_prk](https://docs.rs/hkdf/latest/hkdf/struct.Hkdf.html#method.from_prk) constructor, then use the resulting [Hkdf](https://docs.rs/hkdf/latest/hkdf/struct.Hkdf.html) object as usual.
> 
> Performance is irrelevant here. Why not follow the default way of doing things?

I didn't do it for performance reasons but I read that as the recommended way. If you have a strong key, we can use that. I think not doing that would be the odd thing, no? If I'd come across this code and see that `new` is used despite a strong key, I'd wonder why the author would think that the key wasn't strong enough despite being the node's private key.

Reply:
Do the relatively short ed25519 keys count as strong in this case? What is "long enough" here?

Reply:
It must be at least as long as the output which is 32b for sha256. That is what it is in the source code at least.

Happy to use `new` as well. I didn't anticipate this to steer up this much discussion :)